app/models/concerns/course/levels_concern.rb

```diff
@@ -0,0 +1,50 @@
+module Course::LevelsConcern
+  extend ActiveSupport::Concern
+
+  # Generates the level numbers for the Course::Levels,
+  # since only the thresholds are stored in database.
+  # The entries should have been sorted during the
+  # database query.
+  #
+  # @return [Array<Course::Levels>] Array with numbered Course::Levels
+  def numbered_levels
+    levels.each_with_index.map do |level, index|
+      level.level_number = index + 1
```

Comment:
This doesn't duplicate the level.

`level.tap { |l| l.level_number = index + 1 }`